infra/license/license-checker.ts

```diff
@@ -31,6 +31,129 @@ var args = process.argv.slice(2)
 const inputPathOfLicenseList = args[0];
 const verificationTag = args[1];
 
+const VerificationResultType = {
+  pass: 'No license issue found',
+
+  warnConditionalLicense: 'Additional manual verification is needed',
+  warnNeverChecked: 'Never checked license is detected',
+  warnUnknownLicense: 'License is not found',
+
+  failForbiddenLicense: 'Forbidden license is found',
+  failForbiddenPackage: 'Forbidden package is used'
```

Comment:
optional) `failForbiddenPackage` is a name of property. At the same time, string format of `"failForbiddenPackage"` is used inside code. IMHO, this is error-prone. 
Maybe some kind of enum class representing errror types like `failForbiddenPackage` could be defined and `VerificationResultType` can be defined as a type of `Map<ErrorType, string>`.  :-)

Reply:
> this is error-prone.

- Ref : https://stackoverflow.com/questions/55377365/what-does-keyof-typeof-mean-in-typescript

First of all, `failForbiddenPackage` is not string but literal type(`VerificationResultTypeKeys`).
And even `"failForbiddenPackage"` has string format, they are not string! :D

```typescript
    switch (packageJudgment[pkgName as keyof typeof packageJudgment].permitted) {
      ...
      case 'no':
        return 'abc'; // Error : Type '"abc"' is not assignable to type '"pass" | "warnConditionalLicense" | ... '
        return 'failForbiddenPackage';
      ...
    }
```

> Maybe some kind of enum class representing errror types like failForbiddenPackage could be defined

- Ref : https://engineering.linecorp.com/ko/blog/typescript-enum-tree-shaking/

Enum is not recommended in typescript T.T
And actually, `VerificationResultType` is already type like `Map<VerificationResultTypeKeys, string>` :)

---

I found that I forgot to add `as const` to `VerificationResultType` :)
I will add it!

Reply:
@llFreetimell Wow, I learn good stuff today. Thanks for great information. :-)

Reply:
👍